src/main/scala/edu/twitter/model/impl/neuralnetwork/DataIterator.scala

```diff
@@ -0,0 +1,179 @@
+package edu.twitter.model.impl.neuralnetwork
+
+import java.io.IOException
+import java.util
+import java.util.NoSuchElementException
+
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.Row
+import org.deeplearning4j.models.embeddings.wordvectors.WordVectors
+import org.deeplearning4j.text.tokenization.tokenizer.preprocessor.CommonPreprocessor
+import org.deeplearning4j.text.tokenization.tokenizerfactory.{DefaultTokenizerFactory, TokenizerFactory}
+import org.nd4j.linalg.dataset.DataSet
+import org.nd4j.linalg.dataset.api.DataSetPreProcessor
+import org.nd4j.linalg.dataset.api.iterator.DataSetIterator
+import org.nd4j.linalg.factory.Nd4j
+import org.nd4j.linalg.indexing.{INDArrayIndex, NDArrayIndex}
+
+import scala.collection.JavaConverters._
+
+
+/**
+  * An Iterator used to convert the training (or the testing) data to the format
+  * that the neural network model understands.
+  *
+  * @param data           The labeled tweets
+  * @param wordVectors    WordVectors object
+  * @param batchSize      Size of each minibatch for training
+  * @param truncateLength If reviews exceed
+  */
+@throws[IOException]
+class DataIterator(val data: RDD[Row],
+                   val wordVectors: WordVectors,
+                   val batchSize: Int,
+                   val truncateLength: Int) extends DataSetIterator {
+
+  private final val vectorSize = wordVectors.getWordVector(wordVectors.vocab.wordAtIndex(0)).length
+
+  private val dataList = data.collect()
+  final private val tokenizerFactory: TokenizerFactory = new DefaultTokenizerFactory
+  tokenizerFactory.setTokenPreProcessor(new CommonPreprocessor)
+
+  private var _cursor: Int = 0
+
+  def next(num: Int): DataSet = {
```

Comment:
Can't see why this is necessary, the documentation is already written in the interface and we have no extra details to mention.

Reply:
because this is the only public function I will use when I write it I want the IDE to visualize the doc of this function IDE will not get the doc from the class

Reply:
Done